tephra-core/src/main/java/co/cask/tephra/TransactionManager.java

```diff
@@ -905,6 +906,11 @@ private void doCommit(long writePointer, Set<ChangeId> changes, long commitPoint
     }
     // moving read pointer
     moveReadPointerIfNeeded(writePointer);
+    if (checkpointWritePointers != null) {
+      for (long checkpointWritePointer : checkpointWritePointers) {
```

Comment:
@poornachandra  not sure if we can get away with only calling moveReadPointerIfNeeded for the last check point write pointer (since they are always increasing) ?